parse_test.go

```diff
@@ -49,6 +62,20 @@ message Channel {
 }
 `
 
+const protoWithNestedFieldOptions = `
+syntax = "proto3";
+
+package test;
+
+message Channel {
+  int64 id = 1;
+  string name = 2;
+  string description = 3;
+  map<string, int32> attributes = 4;
+  string address = 5 [(custom_options) = { personal: true internal: false owner: "some owner" }];
```

Comment:
It appears that there is another way to describe these options:

```protobuf
// a field in a message where the the definition isn't a map literal (from the proto docs)
optional int32 a = 1 [(foo_options).opt1 = 123, (foo_options).opt2 = "baz"];
```

Would you mind adding a test for this as well -- otherwise I'm not sure that the parser handles this syntax as well.